org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/callhierarchy/CallerMethodWrapper.java

```diff
@@ -213,4 +213,15 @@ public boolean isExpandWithConstructorsSet() {
 		return fIsExpandWithConstructorsSet;
 	}
 
+	private static boolean isRecordComponent(IMember member) throws JavaModelException {
+		if (member.getElementType() == IJavaElement.FIELD && member instanceof IField f) {
+			if (f.isRecordComponent()) {
+				return true;
+			}
+			if (f.getParent() instanceof IType p && p.isRecord()) {
```

Comment:
For whatever reason, the `SourceType` that is passed to `CallerMethodWrapper.getAccurateSearchScope()` returns `false` for `isRecordComponent()`. When using the reproducer from #571. So we need this second part here.

When selecting the record component (e.g. `wheels` in `Car.java` from the reproducer) `isRecordComponent()` returns `true` as expected.

Unfortunately I don't know how to test with the reproducer from #571. I would need to get into the method body of the method (as done in the IDE), select the method and compute the call hierarchy from there.

Possible, but I'm not sure if its worth the effort. Especially if we have a JDT core bug to fix for `isRecordComponent()`. WDYM @iloveeclipse ?

Reply:
Simeon, please create JDT bug, this soukds really suspicious.

Reply:
Created: https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1035

Reply:
I have a fix for JDT core, will push soon. Just found out that RecordsElementTests were never executed...

Reply:
Simeon, you can remove the workaround now. JDT core fix is merged.